std/utf.d

```diff
@@ -1905,6 +1908,14 @@ unittest
 }
 /* =================== Encode ======================= */
 
+private dchar _utfException(UseReplacementDchar useReplacementDchar)(string msg, dchar c)
```

Comment:
I guess I need to add simillar flag to UtfMatcher in std.uni. Anyway what is the direction of our UTF processing - is it along the lines of: assume that `char[]` is valid and run validation on `ubyte[]` to get `char[]` ?